pp_hot.c

```diff
@@ -169,6 +169,57 @@ PP(pp_padsv_store)
     RETURN;
 }
 
+/* A mashup of simplified AELEMFAST_LEX + SASSIGN OPs */
+
+PP(pp_aelemfastlex_store)
+{
+    dSP;
+    OP * const op = PL_op;
+    SV* const val = TOPs; /* RHS value to assign */
+    AV * const av = MUTABLE_AV(PAD_SV(op->op_targ));
+    const I8 key   = (I8)PL_op->op_private;
+    SV * targ = NULL;
+
+    /* !OPf_STACKED is not handled by this OP */
+    assert(op->op_flags & OPf_STACKED);
+    assert(key >= 0);
+
+    /* Inlined, simplified pp_aelemfast here */
+    assert(SvTYPE(av) == SVt_PVAV);
+    SV** svp;
+
+    /* inlined av_fetch() for simple cases ... */
+    if (!SvRMAGICAL(av) && key >= 0 && key <= AvFILLp(av)) {
```

Comment:
Does this condition need to test `key >= 0`? You already `assert()`ed on that earlier, and rpeep only rewrote it to this opcode if so.

Reply:
Oh yeah, good point. I'll remove that test.